components/managedidentity/locals.tf

```diff
@@ -15,24 +42,13 @@ locals {
     preview = {
       subscription_id = "1c4f0704-a29e-403d-b719-b90c34ef14c9"
     }
-    # DCD-CNP-QA
     ithc = {
       subscription_id = "7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c"
     }
     perftest = {
       subscription_id = "7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c"
     }
-    # DCD-CNP-Prod
-    prod = {
-      subscription_id = "8999dec3-0104-4a27-94ee-6588559729d1"
-    }
-    # DTS-CFTSBOX-INTSVC
-    cftsbox-intsvc = {
-      subscription_id = "1497c3d7-ab6d-4bb7-8a10-b51d03189ee3"
-    }
-    # DTS-CFTPTL-INTSVC
-    cftptl-intsvc = {
-      subscription_id = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
-    }
   }
+
+  valid_envs = contains(["sandbox", "prod"], local.mi_environment)
```

Comment:
This still needs re-looking at, all envs need the role assignment for the monitoring MI besides the ones the job is **not** running in, which I think is ptl/ptlsbox.

Reply:
it is also missing on sbox but there is a cosmosdb there for it.

Reply:
ooh it is missing on sbox too. make me think if should add sbox in var too